scripts/publish/publish.js

```diff
@@ -41,32 +42,53 @@ checkout();
 function changeVersion() {
   log.info('Updating version number...');
 
-  const packageJson = path.join(__dirname, '../../components/package.json');
-  const appComponent = path.join(__dirname, '../site/_site/src/app/app.component.ts') ;
-  const codeBox = path.join(__dirname, '../site/_site/src/app/share/nz-codebox/nz-codebox.component.ts');
-  const currentVersion = fs.readFileSync(packageJson, 'utf-8').match(/"version": "([0-9.]+)"/)[ 1 ];
+  const packageJsonPath = path.join(__dirname, '../../components/package.json');
+  const appComponentPath = path.join(__dirname, '../site/_site/src/app/app.component.ts') ;
+  const codeBoxPath = path.join(__dirname, '../site/_site/src/app/share/nz-codebox/nz-codebox.component.ts');
 
+  const packageJson = fs.readJsonSync(packageJsonPath);
+  const currentVersion = packageJson.version;
   let versionNumberValid = false;
   let version;
 
+  function parseVersion(version) {
+    const matches = version.match(versionNameRegex);
+
+    if (!matches) {
+      return null;
+    }
+
+    return {
+      major: Number(matches[1]),
+      minor: Number(matches[2]),
+      patch: Number(matches[3]),
+      preTag: matches[4],
+      pre: Number(matches[5]),
+    }
+  }
+
   function checkVersionNumber(cur, next) {
     // Must be numbers and dots.
-    if (next.indexOf('rc') > -1) { return true; }
-    if (!/^[1-9][0-9.]{1,10}[0-9]$/.test(next)) { return false; }
+    if (!versionNameRegex.test(next)) { return false; }
+
+    const curVersion = parseVersion(cur);
+    const nextVersion = parseVersion(next);
 
-    const curArr = cur.split('.');
-    const nextArr = next.split('.');
-    const length = curArr.length;
+    for (k of ['major', 'minor', 'patch']) {
+      if (curVersion[k] < nextVersion[k]) {
```

Comment:
It seems that people have different style opinion on the extra whitespace around brackets.

The previous code here is
```ts
if (curArr[ i ] < nextArr[ i ]) { return true; }
```

while the new code does not have these extra whitespaces like `[ i ]`.

It is time-wasting to argue which one is better, but please stick to only one style guide and make sure the style guide is enforced in the whole project without exception.